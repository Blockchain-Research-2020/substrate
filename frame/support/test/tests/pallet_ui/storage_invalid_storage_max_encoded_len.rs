#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::{Hooks, StorageValue};
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::config]
	pub trait Config: frame_system::Config {}

	#[pallet::pallet]
	#[pallet::set_storage_max_encoded_len]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::hooks]
	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}

	#[pallet::call]
	impl<T: Config> Pallet<T> {}

	#[pallet::storage]
	type Foo<T> = StorageValue<_, Vec<u32>>;
}

fn main() {
}
