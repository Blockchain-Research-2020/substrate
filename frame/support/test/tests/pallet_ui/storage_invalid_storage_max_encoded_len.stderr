error[E0277]: the trait bound `Vec<u32>: MaxEncodedLen` is not satisfied
  --> $DIR/storage_invalid_storage_max_encoded_len.rs:10:12
   |
10 |     #[pallet::set_storage_max_encoded_len]
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `MaxEncodedLen` is not implemented for `Vec<u32>`
   |
   = note: required because of the requirements on the impl of `StorageMaxEncodedLen` for `frame_support::pallet_prelude::StorageValue<_GeneratedPrefixForStorageFoo<T>, Vec<u32>>`
   = note: required by `storage_max_encoded_len`
